packages/experimental/TabList/src/TabListAnimatedIndicator/TabListAnimatedIndicator.types.ts

```diff
@@ -0,0 +1,33 @@
+import type { LayoutRectangle, ViewStyle } from 'react-native';
+
+import type { IViewProps } from '@fluentui-react-native/adapters';
+
+export const tablistAnimatedIndicatorName = 'TabListAnimatedIndicator';
+
+export interface TabListAnimatedIndicatorProps extends IViewProps {
+  vertical?: boolean;
+  styles?: {
+    container: ViewStyle;
+    indicator: ViewStyle;
+  };
+}
+
+export interface TabLayoutInfo extends Partial<LayoutRectangle> {
+  startMargin?: number;
+  tabBorderWidth?: number;
+}
+
+export type ListLayoutMap = { [tabKey: string]: TabLayoutInfo };
+
+export interface AnimatedIndicatorStyles {
+  container: ViewStyle;
+  indicator: ViewStyle;
+}
+export type AnimatedIndicatorStylesUpdate = Partial<AnimatedIndicatorStyles>;
+
+export interface AnimatedIndicatorState {
+  addToLayoutMap: (tabKey: string, layout: TabLayoutInfo) => void;
```

Comment:
Hmm... something seems weird to me to have an object named "state" to have functions on it. I feel like this could be an object with styles and layout info and addToLayoutMap and updateStyles could be standalone functions. Neither of the functions depend on component information so they shouldn't need useCallback